test/unit/cursor/aggregation_cursor.test.ts

```diff
@@ -0,0 +1,160 @@
+import { expect } from 'chai';
+
+import { type AggregationCursor, MongoClient } from '../../mongodb';
+
+describe('class AggregationCursor', () => {
+  let client: MongoClient;
+  let cursor: AggregationCursor;
+
+  beforeEach(async function () {
+    client = new MongoClient('mongodb://iLoveJavascript');
+    cursor = client.db().aggregate();
+  });
+
+  afterEach(async function () {
+    await cursor.close();
+    await client.close();
+  });
+
+  context('get pipeline()', () => {
+    it('returns the current aggregation pipeline', () => {
+      expect(cursor.pipeline).to.deep.equal([]);
+    });
+  });
+
+  context('clone()', () => {
+    it('returns a new cursor with a different session', () => {
+      const cloned = cursor.clone();
+      expect(cursor).to.not.equal(cloned);
+      expect(cursor.session).to.not.equal(cloned.session);
+    });
+  });
+
+  context('map()', () => {
+    /*
+     * map does not actually return a new cursor,
+     * the method exists to allow typescript to redefine the output based on the transform
+     */
+    it('returns the same cursor instance', () => {
+      const mappedCursor = cursor.map(() => ({ blah: 1 }));
+      expect(cursor).to.equal(mappedCursor);
+    });
+  });
+
+  context('geoNear()', () => {
+    it('adds a $geoNear stage', () => {
+      cursor.geoNear({ lat: 1, lon: 1 });
+      expect(cursor.pipeline).to.have.deep.property('0', { $geoNear: { lat: 1, lon: 1 } });
+    });
+  });
+
+  context('unwind()', () => {
+    it('adds a $unwind stage', () => {
+      cursor.unwind({ blah: 1 });
+      expect(cursor.pipeline).to.have.deep.property('0', { $unwind: { blah: 1 } });
+    });
+  });
+
+  context('sort()', () => {
+    it('adds a $sort stage', () => {
+      cursor.sort({ _id: -1 });
+      expect(cursor.pipeline).to.have.deep.property('0', { $sort: { _id: -1 } });
+    });
+  });
+
+  context('skip()', () => {
+    it('adds a $skip stage', () => {
+      cursor.skip(2);
+      expect(cursor.pipeline).to.have.deep.property('0', { $skip: 2 });
+    });
+  });
+
+  context('redact()', () => {
+    it('adds a $redact stage', () => {
+      cursor.redact({ redact: true });
+      expect(cursor.pipeline).to.have.deep.property('0', { $redact: { redact: true } });
+    });
+  });
+
+  context('lookup()', () => {
+    it('adds a $lookup stage', () => {
+      cursor.redact({ lookup: true });
```

Comment:
Why aren't we calling `cursor.lookup()` here?

Reply:
fixed, good catch ty!